test/testMinsky.cc

```diff
@@ -1277,3 +1282,613 @@ TEST(TensorOps, evalOpEvaluate)
+      // Should push only if different from previous
+      EXPECT_TRUE(pushed || history.size() > 0);
+      
+      auto var2 = model->addItem(VariablePtr(VariableType::flow, "histVar2"));
+      pushHistory();
+      
+      size_t histSize = history.size();
+      if (histSize > 0)
+      {
+        undo(1);
+        EXPECT_TRUE(model->items.size() <= 3); // May have been restored
+      }
+    }
+
+    // Test dimension operations
+    TEST_F(MinskySuite, dimensionOperations)
+    {
+      // Create a dimension with proper Type enum (e.g., Dimension::value)
+      dimensions.emplace("testDim", Dimension(Dimension::value, "test"));
+      
+      // Create a variable with this dimension in its hypercube
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "dimTestVar"));
+      auto hc = variableValues[":dimTestVar"]->tensorInit.hypercube();
+      hc.xvectors.emplace_back("testDim", Dimension(Dimension::value, "test"));
+      variableValues[":dimTestVar"]->tensorInit.hypercube(hc);
+      
+      renameDimension("testDim", "newTestDim");
+      
+      // Verify dimension was renamed in dimensions map
+      EXPECT_GT(dimensions.count("newTestDim"), 0);
+      EXPECT_EQ(dimensions.count("testDim"), 0);
+      
+      // Verify variableValue hypercube was also updated
+      auto updatedHc = variableValues[":dimTestVar"]->tensorInit.hypercube();
+      bool foundRenamed = false;
+      bool foundOld = false;
+      for (const auto& xv : updatedHc.xvectors) {
+        if (xv.name == "newTestDim") foundRenamed = true;
+        if (xv.name == "testDim") foundOld = true;
+      }
+      EXPECT_TRUE(foundRenamed);
+      EXPECT_FALSE(foundOld);
+      
+      dimensions.clear();
+    }
+
+    // Test Godley operations
+    TEST_F(MinskySuite, godleyOperations)
+    {
+      auto g1 = new GodleyIcon;
+      model->addItem(g1);
+      g1->table.resize(3, 3);
+      g1->table.cell(0,1) = "stock1";
+      g1->table.cell(2,1) = "flow1";
+      g1->update();
+      
+      setAllDEmode(true);
+      EXPECT_TRUE(g1->table.doubleEntryCompliant);
+      
+      setAllDEmode(false);
+      EXPECT_FALSE(g1->table.doubleEntryCompliant);
+      
+      vector<string> flowVars = allGodleyFlowVars();
+      EXPECT_GT(flowVars.size(), 0);
+    }
+
+    // Test variable type conversion
+    TEST_F(MinskySuite, convertVarType)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "convertVar"));
+      EXPECT_EQ(VariableType::flow, variableValues[":convertVar"]->type());
+      
+      convertVarType(":convertVar", VariableType::parameter);
+      EXPECT_EQ(VariableType::parameter, variableValues[":convertVar"]->type());
+    }
+
+    // Test addIntegral
+    TEST_F(MinskySuite, addIntegral)
+    {
+      auto var1 = model->addItem(VariablePtr(VariableType::flow, "integVar"));
+      canvas.item = var1;
+      
+      size_t itemsBefore = model->items.size();
+      addIntegral();
+      EXPECT_GT(model->items.size(), itemsBefore);
```

Comment:
Should also check that var1's type is integral, and also that its is attached to an integral by referenciung the controller attribute of variable,